docs/user-guide/coordinate-transformations.ipynb

```diff
@@ -0,0 +1,359 @@
+    "It automatically handles:\n",
+    "- Renaming of dimensions, if dimension-coordinates are transformed.\n",
+    "- Change of coordinates to attributes to avoid interference of coordinates consumed by the transformation in follow-up operations.\n",
+    "- Conversion of event-coordinates of binned data, if present.\n",
+    "\n",
+    "### Basic example\n",
+    "\n",
+    "We start by revisiting the example given in [Motivation](#Motivation).\n",
+    "The building blocks `transform_coords` operates on are functions with named parameters.\n",
+    "The parameter names define the names of the *input coordinates to consume*.\n",
+    "Let us define `x_square`, which will *consume* `x`:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "be50be41-9486-4943-99da-99653639ea98",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def x_square(x):\n",
+    "    return x * x"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f22ec1c0-ad99-418b-820a-a6dad6fb82aa",
+   "metadata": {},
+   "source": [
+    "Next, we create a `dict`, mapping from an output coord name to a function that can create this coordinate.\n",
+    "The [sc.show_graph](../generated/functions/scipp.show_graph.rst#scipp.show_graph) helper is a convient tool for visualizing the coordinate transformation defined by such as mapping:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "33f5e9ef-35a8-49e6-9731-8b33a525c8fb",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "graph = {'x^2':x_square}\n",
+    "sc.show_graph(graph)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a658c4ce-ab11-41ed-84c5-af8df71838ab",
+   "metadata": {},
+   "source": [
+    "Here, the `x` coordinate can be consumed by the `x_square` function, creating the `x^2` coordinate.\n",
+    "Note that the function name and coordinate are unrelated.\n",
+    "Next, we can call `transform_coords`.\n",
+    "Apart from the graph, we also pass a list of desired output coordinates, here simply `['x^2']`.\n",
+    "`transform_coords` returns a new (shallow-copied) data array with added coordinates:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ba462ced-c928-4f02-b60d-af8a602c03dd",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "transformed = da.transform_coords(['x^2'], graph=graph)\n",
+    "transformed"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ab752aee-f2d7-4cae-8c7a-b33a2f7e8033",
+   "metadata": {},
+   "source": [
+    "Note how `x` is now an attribute, i.e., operations will not use it for alignment anymore.\n",
+    "This is important since it will allow for operations combining `transformated` with other data that may have matching `x^2` but not `x`.\n",
```

Comment:
```suggestion
    "This is important since it will allow for operations combining `transformed` with other data that may have matching `x^2` but not `x`.\n",
```